packages/web3-eth-ens/types/index.d.ts

```diff
@@ -0,0 +1,157 @@
+ * @date 2018
+ */
+
+import * as net from 'net';
+import {
+    AbstractWeb3Module,
+    AccountsBase,
+    NetworkBase,
+    PromiEvent,
+    provider,
+    TransactionConfig,
+    TransactionSigner,
+    Web3ModuleOptions
+} from 'web3-core';
+import { formatters } from 'web3-core-helpers';
+import { AbiCoder } from 'web3-eth-abi';
+import { Contract, ContractModuleFactory } from 'web3-eth-contract';
+import { Utils } from 'web3-utils';
+
+export class Ens extends AbstractWeb3Module {
+    constructor(
+        provider: provider,
+        net?: net.Socket | null,
+        accounts?: AccountsBase | null,
+        options?: Web3ModuleOptions
+    );
+
+    registry: Registry;
+    transactionSigner: TransactionSigner;
+
+    resolver(name: string): Promise<Contract>;
+
+    supportsInterface(
+        name: string,
+        interfaceId: string,
+        callback?: (error: Error, supportsInterface: boolean) => void
+    ): Promise<boolean>;
+
+    getAddress(
+        name: string,
+        callback?: (error: Error, address: string) => void
+    ): Promise<string>;
+
+    setAddress(
+        name: string,
+        address: string,
+        sendOptions: TransactionConfig,
+        callback?: (error: Error, result: any) => void
+    ): PromiEvent<any>;
+
+    getPubkey(
+        name: string,
+        callback?: (error: Error, result: { [x: string]: string }) => void
+    ): Promise<{ [x: string]: string }>;
+
+    setPubkey(
+        name: string,
+        x: string,
+        y: string,
+        sendOptions: TransactionConfig,
+        callback?: (error: Error, result: any) => void
+    ): PromiEvent<any>;
+
+    getText(
+        name: string,
+        key: string,
+        callback?: (error: Error, ensName: string) => void
+    ): Promise<string>;
+
+    setText(
+        name: string,
+        key: string,
+        value: string,
+        sendOptions: TransactionConfig,
+        callback?: (error: Error, result: any) => void
+    ): PromiEvent<any>;
+
+    getContent(
+        name: string,
+        callback?: (error: Error, contentHash: string) => void
+    ): Promise<string>;
+
+    setContent(
+        name: string,
+        hash: string,
+        sendOptions: TransactionConfig,
+        callback?: (error: Error, result: any) => void
+    ): PromiEvent<any>;
+
+    getMultihash(
+        name: string,
+        callback?: (error: Error, multihash: string) => void
+    ): Promise<string>;
+
+    setMultihash(
+        name: string,
+        hash: string,
+        sendOptions: TransactionConfig,
+        callback?: (error: Error, result: any) => void
+    ): PromiEvent<any>;
+
+    getContenthash(
+        name: string,
+        callback?: (error: Error, contenthash: string) => void
+    ): Promise<string>;
+
+    setContenthash(
+        name: string,
+        hash: string,
+        sendOptions: TransactionConfig,
+        callback?: (error: Error, result: any) => void
+    ): PromiEvent<any>;
+}
+
+export class Registry {
+    constructor(
```

Comment:
It would be great if you could check the 1.0 constructor of the Registry object. The constructor will be different.